memory_wrapper.h

```diff
@@ -20,7 +20,7 @@ class memory_wrapper {
   static constexpr int kEntryBits = kTotalBits - kOffsetBits;
   static constexpr int kEntryMask = 0x0FFF;
   static constexpr int kMapEntry = 1 << kEntryBits;
-  static constexpr size_t kMaxEntry = 1l << kTotalBits;
+  static constexpr size_t kMaxEntry = ((1ull << 32) - 1);
```

Comment:
A long is not guaranteed to be 64-bit, so I changed this to an unsigned long long. Furthermore, a size_t shifted << 32 on 32-bit systems will be null (and when tested, caused a crash). I subtract one to fix this but the size_t could be changed to uint64_t (and updated elsewhere, where that would conflict).